src/server/active_tran_server.cpp

```diff
@@ -110,15 +173,27 @@ active_tran_server::connection_handler::receive_saved_lsa (page_server_conn_t::s
   assert (sizeof (log_lsa) == message.size ());
   std::memcpy (&saved_lsa, message.c_str (), sizeof (log_lsa));
 
-  if (log_Gl.m_max_ps_flushed_lsa < saved_lsa)
-    {
-      log_Gl.update_max_ps_flushed_lsa (saved_lsa);
-    }
+  assert (saved_lsa > get_saved_lsa ()); // increasing monotonically
+  set_saved_lsa (saved_lsa);
 
   if (prm_get_bool_value (PRM_ID_ER_LOG_COMMIT_CONFIRM))
     {
       _er_log_debug (ARG_FILE_LINE, "[COMMIT CONFIRM] Received LSA = %lld|%d.\n", LSA_AS_ARGS (&saved_lsa));
     }
+
+  log_Gl.update_ps_consensus_flushed_lsa ();
```

Comment:
Consensus calculation has the potential of being a costly calculation (mutex and all..)
Here we have the option to:
- calculate the consensus on the communication thread (as done now) - and then notify the waiting client threads
- just notify the waiting client threads and let one of them (the first one able to acquire that mutex) do it

Which one do you think is the best option?

Reply:
The second seems better in that the waiting threads are just waiting and the request handler can be busy. I'll try it. Thank you.